pkg/operators/eksnodegroup/operator.go

```diff
@@ -0,0 +1,270 @@
+				name = resource.Name
+				namespace = resource.Namespace
+			}
+		}
+	}
+
+	if name != "" && namespace != "" {
+		annotations := map[string]string{
+			"StackID":      msg.ParsedMessage["StackId"],
+			"StackName":    msg.ParsedMessage["StackName"],
+			"ResourceType": msg.ParsedMessage["ResourceType"],
+		}
+		if msg.ParsedMessage["ResourceStatus"] == "ROLLBACK_COMPLETE" {
+			obj, err := deleteStack(config, name, namespace, msg.ParsedMessage["StackId"])
+			if err != nil {
+				return err
+			}
+			config.Recorder.AnnotatedEventf(obj, annotations, corev1.EventTypeWarning, strcase.ToCamel(strings.ToLower(msg.ParsedMessage["ResourceStatus"])), msg.ParsedMessage["ResourceStatusReason"])
+		} else if msg.ParsedMessage["ResourceStatus"] == "DELETE_COMPLETE" {
+			obj, err := updateStatus(config, name, namespace, msg.ParsedMessage["StackId"], msg.ParsedMessage["ResourceStatus"], msg.ParsedMessage["ResourceStatusReason"])
+			if err != nil {
+				return err
+			}
+			config.Recorder.AnnotatedEventf(obj, annotations, corev1.EventTypeWarning, strcase.ToCamel(strings.ToLower(msg.ParsedMessage["ResourceStatus"])), msg.ParsedMessage["ResourceStatusReason"])
+			err = incrementRollbackCount(config, name, namespace)
+			if err != nil {
+				return err
+			}
+		} else {
+			obj, err := updateStatus(config, name, namespace, msg.ParsedMessage["StackId"], msg.ParsedMessage["ResourceStatus"], msg.ParsedMessage["ResourceStatusReason"])
+			if err != nil {
+				return err
+			}
+			config.Recorder.AnnotatedEventf(obj, annotations, corev1.EventTypeNormal, strcase.ToCamel(strings.ToLower(msg.ParsedMessage["ResourceStatus"])), msg.ParsedMessage["ResourceStatusReason"])
+		}
+
+	}
+
+	return nil
+}
+
+func (c *Operator) onAdd(obj interface{}) {
+	s := obj.(*awsV1alpha1.EKSNodeGroup).DeepCopy()
+	if s.Status.ResourceStatus == "" || s.Status.ResourceStatus == "DELETE_COMPLETE" {
+		cft := New(c.config, s, c.topicARN)
+		output, err := cft.CreateStack()
+		if err != nil {
+			c.config.Logger.WithError(err).Errorf("error creating eksnodegroup '%s'", s.Name)
+			return
+		}
+		c.config.Logger.Infof("added eksnodegroup '%s' with stackID '%s'", s.Name, string(*output.StackId))
+		c.config.Logger.Infof("view at https://console.aws.amazon.com/cloudformation/home?#/stack/detail?stackId=%s", string(*output.StackId))
+
+		_, err = updateStatus(c.config, s.Name, s.Namespace, string(*output.StackId), "CREATE_IN_PROGRESS", "")
+		if err != nil {
+			c.config.Logger.WithError(err).Error("error updating status")
+		}
+	}
+}
+
+func (c *Operator) onUpdate(oldObj, newObj interface{}) {
+	oo := oldObj.(*awsV1alpha1.EKSNodeGroup).DeepCopy()
+	no := newObj.(*awsV1alpha1.EKSNodeGroup).DeepCopy()
+
+	if no.Status.ResourceStatus == "DELETE_COMPLETE" {
+		c.onAdd(no)
+	}
+	if helpers.IsStackComplete(oo.Status.ResourceStatus, false) && !reflect.DeepEqual(oo.Spec, no.Spec) {
+		cft := New(c.config, oo, c.topicARN)
+		output, err := cft.UpdateStack(no)
```

Comment:
If CA has changed the current size of the ASG out of band of CFN stack, can calling update undo that change?